### ISSUE TITLE
backward sync is supposed to bring forward the background buffer or tab?

### ISSUE THREAD

**Issue by Kansasmanjaro** [lervag/vimtex#2800]
Say that I have b.tex as the current buffer in the nvim window, with b.tex as background buffer. I thought if I C+left click on a.pdf will bring forward the buffer  a.tex, but this is not the case;  b.tex stays in the main window and the source code of the a.pdf isn't brought forward to the nvim window.
Doesn't vimtex already has this feature? Or somthing is wrong with my setup?

The following is my setup in vim and nothing is here for zathura except `set synctex true`

```
let g:vimtex_view_general_viewer = 'zathura'
let g:vimtex_view_method = 'zathura'
let g:vimtex_quickfix_mode = 0
```

**Comment by lervag** [comment]
> Say that I have b.tex as the current buffer in the nvim window, with b.tex as background buffer. I thought if I C+left click on a.pdf will bring forward the buffer a.tex, but this is not the case; b.tex stays in the main window and the source code of the a.pdf isn't brought forward to the nvim window. Doesn't vimtex already has this feature? Or somthing is wrong with my setup?

Can you be even more specific in describing your issue? I.e., specify the content of `a.tex` and `b.tex`, and show the precise steps to reproduce this. E.g. something like this:

```tex
% a.tex
\documentclass{minimal}
\begin{document}
Hello A
\end{document}

% b.tex
\documentclass{minimal}
\begin{document}
Hello B
\end{document}
```

1. Open a terminal
2. `cd path/to/test/folder`
3. `vim a.tex`
4. `\ll`
5. wait until the viewer (Zathura) is opened
6. `:e b.tex`
7. Control+left click to perform inverse search from Zathura

> The following is my setup in vim and nothing is here for zathura except `set synctex true`
> 
> ```
> let g:vimtex_view_general_viewer = 'zathura'
> let g:vimtex_view_method = 'zathura'
> let g:vimtex_quickfix_mode = 0
> ```

You don't need `let g:vimtex_view_general_viewer` here, so just remove that. Also, how do you install VimTeX? Which plugin manager and with which code?

**Comment by Kansasmanjaro** [comment]
Thanks the help. Glad to know that vimtex has this already.
I installed Vimtex using vim plug via `Plug 'lervag/vimtex'.

I followed your suggestion and made a folder and tex files.
Did the folloiwng:

1. Open a terminal
2. cd path/to/test/folder
3. nvim a.tex
4. \ll
5. wait until the viewer (Zathura) is opened
6 :e b.tex so that the nvim has b.tex as the the active buffer(now a.tex is in the background)
7 Now Control+left click on a.pdf to perform inverse search from Zathura to a.tex

Result: a.tex was not brought forward. Nvim still has the b.tex as the active buffer.

I did the test with the following init.vim, no other config,
```
let &packpath = &runtimepath
set mouse=a
let g:vimtex_view_method = 'zathura'
let g:vimtex_quickfix_mode = 0
call plug#begin()
Plug 'lervag/vimtex'
call plug#end()
```

**Comment by ejmastnak** [comment]
For what it's worth I have the same behavior as @Kansasmanjaro (with Neovim/Zathura/Arch; happy to post details). I'd always thought this (i.e. not bringing the background buffer to foreground) was the expected behavior and am perfectly satisfied as is, but happy to help investigate if another data point is of any help.

**Comment by lervag** [comment]
Ah, yes, it seems I was wrong. The logic is all here:

https://github.com/lervag/vimtex/blob/a59a49b54ae039f1a439c9a16c2ea7b461b8e925/autoload/vimtex/view.vim#L74-L80

So: Inverse search works by inspecting all Vim (or neovim) servers available. For each server, we inspect the _currently_ open file in the _current_ window/buffer. If that file is part of LaTeX project that contains the target source file, then we will perform the inverse search. If not, then we abort and do nothing.

We _could_ make this more complex by e.g. looking through the buffers in each Vim/neovim server. But I'm not convinced that it is a good idea. If you were to have two neovim sessions open and both of them had a relevant buffer, then both of them would be activated, regardless of which file and location they had opened. I think this can be confusing. Instead, it seems safer to assume that the user wants to search in the Vim/neovim session that already has a relevant buffer open.

In conclusion: I'm sorry I claimed this "worked", when it in fact does not. However, it does not work because I once decided that it was better if it did not work that way.

**Comment by GuannanHu** [comment]
Thank you for clearing up my confusion and for this great plugin.

**Comment by lervag** [comment]
Glad to help, @GuannanHu, and thanks for the kind words. Just to avoid any confusion: you are not the same person as @Kansasmanjaro?

**Comment by Kansasmanjaro** [comment]
That makes sense. Thank a lot!!
> Ah, yes, it seems I was wrong. The logic is all here:
> 
> https://github.com/lervag/vimtex/blob/a59a49b54ae039f1a439c9a16c2ea7b461b8e925/autoload/vimtex/view.vim#L74-L80
> 
> So: Inverse search works by inspecting all Vim (or neovim) servers available. For each server, we inspect the _currently_ open file in the _current_ window/buffer. If that file is part of LaTeX project that contains the target source file, then we will perform the inverse search. If not, then we abort and do nothing.
> 
> We _could_ make this more complex by e.g. looking through the buffers in each Vim/neovim server. But I'm not convinced that it is a good idea. If you were to have two neovim sessions open and both of them had a relevant buffer, then both of them would be activated, regardless of which file and location they had opened. I think this can be confusing. Instead, it seems safer to assume that the user wants to search in the Vim/neovim session that already has a relevant buffer open.
> 
> In conclusion: I'm sorry I claimed this "worked", when it in fact does not. However, it does not work because I once decided that it was better if it did not work that way.
